### ISSUE TITLE
function --on-variable LINES not called when $LINES changes due to terminal resize

### ISSUE THREAD

**Issue by yuri-kilochek** [fish-shell/fish-shell#2316]
I do

``` fish
function foo --on-variable LINES
    echo "LINES changed"
end
```

and then resize terminal so that `LINES` is changed (i confirm this by manually echoing it). `foo`, however, is never called.

This doesn't seem like https://github.com/fish-shell/fish-shell/issues/1929,  because my fish is 2.2.0, where it is fixed and it indeed works for variables I can change myself via `set`.

**Comment by faho** [comment]
I can confirm that the function isn't called. My guess is that "LINES" is special somehow.

The question though is: What are you trying to achieve? Is that solvable via a signal handler for the "WINCH" signal instead, like

```fish
function foo --on-signal WINCH
    echo "LINES changed $LINES"
end
```

**Comment by yuri-kilochek** [comment]
@faho, i'm trying to make prompt always appear at the bottom, via `tput cup $LINES`, so yes, `WINCH` workaround does work.

**Comment by andrex-e-schulman** [comment]
Same is true for the COLUMNS variable, by the way, which is also reset on terminal resize.

**Comment by zanchey** [comment]
The problem is that `LINES` and `COLUMNS` are "electric" variables - their value is dynamically calculated and never stored, so fish doesn't try to detect when they have changed.

**Comment by faho** [comment]
@zanchey: Could we do that when SIGWINCH is fired? Creating a an on-variable listener seems quite intuitive.

**Comment by ridiculousfish** [comment]
The config scripts use WINCH:

    function __fish_winch_handler --on-signal WINCH
                commandline -f repaint
        end

**Comment by krader1961** [comment]
This has already been fixed as a side-effect of my fixes for other problems with handling the terminal size. Specifically, commit 2be1288ca to fix issue #3740.
